sklearn/utils/estimator_checks.py

```diff
@@ -290,6 +298,9 @@ def check_pipeline_consistency(name, Estimator):
                       random_state=0, n_features=2, cluster_std=0.1)
     X -= X.min()
     y = multioutput_estimator_convert_y_2d(name, y)
+    if name == "SpectralEmbedding":
+        # This data makes sure that the graph is fully connected.
+        X, y = S_X, S_y
```

Comment:
Why a special case in this common check? Once the remaining sign switch is fixed (the constant abs values case), `SpectralEmbedding` should be fully deterministic and there would be no need for a specific dataset in the checks, right?